### PR TITLE
[bug] Remove invalid connection from the idle queue when request new connection

### PR DESCRIPTION
**Pull request by lopn**
[bug] 移除无效连接 ，防止由于无效连接导致 连接总数超出MaxConnPoolSize，导致创建session失败

Fail to create a new session from connection pool, username: xxx, password: xxxx, failed to get connection: No valid connection in the idle queue and connection number has reached the pool capacity

### CODE REVIEW

**Comment by lopn** [comment]
https://discuss.nebula-graph.com.cn/t/topic/9528/10

**Review by Aiee — COMMENTED** [review]
Good job. Please fix the format using `go fmt`.

File: connection_pool.go [hunk 1]
```diff
@@ -144,12 +144,16 @@ func (pool *ConnectionPool) getIdleConn() (*connection, error) {
...
-		for ele := pool.idleConnectionQueue.Front(); ele != nil; ele = ele.Next() {
+		var tmpNextEle *list.Element = nil
+		for ele := pool.idleConnectionQueue.Front(); ele != nil; ele = tmpNextEle {
 			// Check if connection is valid
 			if res := ele.Value.(*connection).ping(); res {
 				newConn = ele.Value.(*connection)
 				newEle = ele
 				break
+			}else{
+				tmpNextEle = ele.Next()
```
Comment (Aiee) [inline connection_pool.go:155]: Good job. Please fix the format using go fmt.

**Review by Aiee — APPROVED** [review]
LGTM.

**Pull request merged**